### PR TITLE
Add V1xGenerator and V3xGenerator.

### PR DESCRIPTION
**Pull request by rturrado**
As subclasses of Generator. With slightly different implementations.

Update C++ version to C++23.

Remove `build_tests` option from conan profiles.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1,105 +1,144 @@
...
+    generator->generate_const_scalar_op("operator+", 's', "ss", "a + b");
+    generator->generate_const_scalar_op("operator//", 'i', "ii", "div_floor(a, b)");
+
+    // Bitwise operators
+    generator->generate_const_scalar_op("operator>>>", 'i', "ii", "(int64_t)(((uint64_t)a) >> b)");
+}
+
+
+void test_v1x(const std::string &header_file_path, const std::string &source_file_path) {
+    std::unique_ptr<func_gen::Generator> generator{};
```
Comment (pablolh) [inline src/main.cpp]: you don't have to use a pointer I believe?
create the generator on the stack, and pass a reference in generate_common_functions and generate_v1x_specific_functions instead

File: src/main.cpp [hunk 2]
```diff
@@ -1,105 +1,144 @@
...
+    // Check command line
+    if (argc != 4) {
+        print_usage();
+        return 1;
+    }
+    
+    // Check version, and test v1x or v3x generators
+    std::string_view version{ argv[3] };
+    if (version == "v1x") {
+        test_v1x(argv[1], argv[2]);
```
Comment (pablolh) [inline src/main.cpp]: why do these functions have `test` in the name?

Comment (rturrado) [reply src/main.cpp]: Thanks Pablo, that's a good point. I've renamed them to `generate_v1x_files` and `generate_v3x_files`.

File: src/main.cpp (hunk 1 above)
Comment (rturrado) [reply src/main.cpp]: Yes, good point, thanks! Done.

I tend to think that I always have to use pointers and dynamic memory to work with runtime polymorphism.

**Review by pablolh — APPROVED** [review]
looks good to me

**Pull request merged**